source/core/NumericString.cpp

```diff
@@ -1496,7 +1496,10 @@ Boolean TypedScanString(SubString* inputString, IntIndex* endToken, const Format
         }
         in.AliasAssign(in.Begin(), in.Begin() + std::min(in.Length(), formatOptions->MinimumFieldWidth + leadingSpace));
     }
-    truncateInput.Append(&in);
+    if (in.Length() <= truncateInput.Capacity())
```

Comment:
`if (in.Length() <= truncateInput.Capacity())`
It does not seem like a good idea to check for something that Append(&in) already does and returns false. I would do something like this:
`
    if (!truncateInput.Append(&in))
        truncateInput.Append(&in, truncateInput.Capacity());
`
Or, even something like this:
`
    truncateInput.Append(&in, std::min(in.Length(), truncateInput.Capacity()));
`